eng/pipelines/templates/steps/install-dotnet.yml

```diff
@@ -27,7 +27,7 @@ steps:
       performMultiLevelLookup: true
       version: "6.0.x"
   - task: UseDotNet@2
-    condition: and(succeeded(), ne(variables['Agent.OS'], 'Windows_NT')) # Windows supports MultiLevelLookup and doesn't need explicit framework installation
+    condition: and(succeeded()) # .NET Core 3.1 does not exist on Windows hosted machine anymore, we still need it for Track 1 SDK
```

Comment:
To help unblock the current release let's put the windows install of this 3.1 directly in the ci.mgmt.yml file to scope it to only track 1 mgmt. Once that is done we can continue to remove this entire step from install-dotnet.yml, happening in https://github.com/Azure/azure-sdk-for-net/pull/35151. Once that happens we can keep the 3.1 install for all OS's in the ci.mgmt.yml file temporarily. We should continue to work on https://github.com/Azure/azure-sdk-for-net/pull/35178 to break this dependency on 3.1 before we can no longer use this EoL framemwork.

Reply:
Added a step in .NET SDK Track 1 CI to install .NET core 3.1 for all OS, so that we can continue with #35174 for the rest.